astropy/cosmology/flrw.py

```diff
@@ -1485,6 +1448,56 @@ def arcsec_per_kpc_proper(self, z):
         return u.arcsec / (self.angular_diameter_distance(z).to(u.kpc) *
                            arcsec_in_radians)
 
+    # -----------------------------------------------------
+
+    def equivalent(self, other):
+        # Note! this method is superseded by FlatFLRWMixin.equivalent in
+        # subclassses of FlatFLRWMixin.
+
+        # case (3) iff FlatFLRWMixin, so delegate check to FlatFLRWMixin
+        if isinstance(other, FlatFLRWMixin):
```

Comment:
Might be nice to avoid having such "knowledge" of other classes. Would it be an idea to use the same trick python does for `__eq__`, i.e., return `NotImplemented` for a given class (and if both do so, it is `False`)?

Reply:
I like that. Hopefully will be much simpler. I'll look into whether this can work here.

Reply:
Works great!